src/messages/MOSDPGRecoveryDeleteReply.h

```diff
@@ -43,7 +49,9 @@ struct MOSDPGRecoveryDeleteReply : public MOSDFastDispatchOp {
   void encode_payload(uint64_t features) override {
     ::encode(pgid.pgid, payload);
     ::encode(map_epoch, payload);
-    ::encode(min_epoch, payload);
+    if (HAVE_FEATURE(features, SERVER_LUMINOUS)) {
```

Comment:
actually wait: if we are a luminous osd, and the remote peer is jewel, we need to either use version == 2 and include the field, or use version == 1 and exclude it.. right?

Reply:
i think we need header.version = 2; } else { header.version = 1; }

Reply:
no, the idea is jewel keeps its current encoding - otherwise we'd need some kind of flag like SERVER_LUMINOUS to tell when to send the v2 encoding or not to jewel

Reply:
hmm, let me double check that - shouldn't have passed my local tests if that was necessary though...

Reply:
Yeah, it's not necessary - since we're not changing compat version it doesn't affect decoding.
If you want to think of 'including min_epoch' as v2 logically that would kind of make sense, except that v1 was both encodings... it doesn't affect correctness though.

We've got 5 possible version pairs:
jewel + 12.2.0 (both v1, incompatible since only 12.2.0 includes min_epoch)
jewel + 12.2.x (v1 and v2, both encode without min_epoch since SERVER_LUMINOUS is missing for jewel)
12.2.0 + 12.2.x (both v2, SERVER_LUMINOUS so both include min_epoch)
12.2.0 + master (v1 and v2, both include min_epoch)
12.2.x + master (both v2, SERVER_LUMINOUS so both use min_epoch)

Reply:
I see.. so the jewel version isn't changing at all then.  Sounds good!